packages/components-vue/projects/vue-wrapper/tests/unit/specs/utils.spec.ts

```diff
@@ -0,0 +1,103 @@
+import * as Vue from 'vue';
+
+describe('getPrefixedTagName()', () => {
+  it('should call inject() with correctParameters', () => {
+    const spy = jest.spyOn(Vue, 'inject').mockReturnValue('');
+
+    getPrefixedTagName('p-text');
+
+    expect(spy).toBeCalledWith(prefixInjectionKey);
+  });
+
+  it('should call throw error if prefix is undefined', () => {
+    jest.spyOn(Vue, 'inject').mockReturnValue(undefined);
+
+    let error;
+    try {
+      getPrefixedTagName('p-text');
+    } catch (e) {
+      error = e;
+    }
+
+    expect(error).toBeDefined();
+  });
+
+  it('should return tagName if no prefix is defined', () => {
+    jest.spyOn(Vue, 'inject').mockReturnValue('');
+    const tagName = 'p-text';
+    getPrefixedTagName('p-text');
+
+    expect(getPrefixedTagName(tagName)).toBe(tagName);
+  });
+
+  it('should return prefixed tagName if prefix is defined', () => {
+    const prefix = 'my-prefix';
+    jest.spyOn(Vue, 'inject').mockReturnValue(prefix);
+    const tagName = 'p-text';
+
+    expect(getPrefixedTagName(tagName)).toBe(prefix + '-' + tagName);
+
+    const prefix2 = 'another-prefix'
+    jest.spyOn(Vue, 'inject').mockReturnValue(prefix2);
+
+    expect(getPrefixedTagName(tagName)).toBe(prefix2 + '-' + tagName);
+  });
+});
+
+describe('syncProperties()', () => {
+  it('should sync passed properties with passed element', () => {
+    const props: {
+      customProp1: string;
+      customProp2: boolean;
+      customProp3: number;
+      customProp4: {};
+    } & Partial<HTMLElement> = { customProp1: 'some prop', customProp2: true, customProp3: 1, customProp4: {} };
+
+    const element = document.createElement('custom-el');
+
+    // TODO: Resolve any casting
+
+    expect((element as any).customProp1).toBeUndefined();
+    expect((element as any).customProp2).toBeUndefined();
+    expect((element as any).customProp3).toBeUndefined();
+    expect((element as any).customProp4).toBeUndefined();
+
+    syncProperties(element, props);
+
+    expect((element as any).customProp1).toBe(props.customProp1);
+    expect((element as any).customProp2).toBe(props.customProp2);
+    expect((element as any).customProp3).toBe(props.customProp3);
+    expect((element as any).customProp4).toBe(props.customProp4);
+  });
+});
+
+describe('addEventListenerToElementRef()', () => {
+  it('should call addEventListener() with correct parameters on passed element', () => {
+    const element = document.createElement('custom-el');
+    const eventName = 'someEventName';
+    const spy = jest.spyOn(element, 'addEventListener');
+
+    addEventListenerToElementRef(element, eventName, () => {});
+
+    expect(spy).toBeCalledWith(eventName, expect.any(Function));
+  });
+
+  it('should call passed emit() with correct parameters', () => {
+    const element = document.createElement('custom-el');
+    const eventName = 'someEventName';
+    const emit = jest.fn();
+
+    const event = new CustomEvent(eventName, { detail: 'someDetail' });
```

Comment:
Reorder so that stuff that belongs together is defined/executed after each other.